### ISSUE TITLE
[Question] Handle Single Sign On into front end app

### ISSUE THREAD

**Issue by Jonathan-Garber** [wpengine/faustjs#1108]
I am trying to sort out a way to handle a SSO into the front end app. This is getting complex since the front end authenticates to WP.

I have been working on having the WP backend authenticate the user and send them across to the front end with a valid authorization code but this doesn't work 100%. It works for the landing page they land on but won't create the cookie on the front end app to keep them authenticated.

Any pointer or direction here?

**Comment by theodesp** [comment]
Hello @Jonathan-Garber. Could you explain a bit more about your setup and your SSO requirements? Are you using a Single Sign On plugin for WordPress or you are having issues with an SSO provider like OKTA?

Have you explored the https://nextjs.org/docs/authentication page as Faust is primarily and Next.js framework?

**Comment by Jonathan-Garber** [comment]
Its an SSO from a third-party using their own solution. I am able to generate a valid authorization code on the WP backend and I can send that across to the front end but I am not sure how to tell the front end to create the authentication cookie properly when presented with a valid code. I'm slowly picking apart the faust library and methods to understand how its doing things.

I was trying to catch the code in middleware and have the app go through the process of creating the authentication cookie and letting the user in.

Right now when the app see's the valid code it will allow authenticated queries and will work but only for the page landed on with the code param in the url. No auth cookie is created and the minute the user browsers away to another app page the code is lost and they are no longer authenticated. I had expected once the code was processed from the url it would pop it off the url and store an authentication cookie.

Any pointers to any methods usable on server-side from the faust library here? I've tried ensureAuthorization, fetchAccessToken, etc.

**Comment by Jonathan-Garber** [comment]
Oh...
I think I found the issue. Testing further to confirm. But I didn't notice until now that I was catching everything from an SSL to non SSL connection which is blocking any attempt to set the authentication cookie (duh)

**Comment by Jonathan-Garber** [comment]
Commenting to Close and to help if anyone else ever hits this.

The issue was indeed as stated above. I've got a mix of settings here between localhost, BrowserStack Local Domain, WP Backend domain etc. I didn't notice I was redirecting from the WP backend after SSO verification to the front end under non-ssl which caused the cookie to never set.

It seems once I fixed my redirect location from WP to App the cookie is now set properly and subsequent navigation in app is fully authenticated.

@theodesp Thanks for posting your advice. I appreciate you taking time to help.
